netsim/modules/vlan.py

```diff
@@ -1280,6 +1280,29 @@ def link_pre_transform(self, link: Box, topology: Box) -> None:
   def module_pre_link_transform(self, topology: Box) -> None:
     create_loopback_vlan_links(topology)
 
+  # We need to copy a few attributes from node VLAN data into access link interface data, for example
+  # the node IPv4/IPv6 address (to support static addressing) and "gateway" attribute to enable
+  # selective VLAN gateway module activation
+  #
+  # This operation needs to be done when the link data structures have already been transformed into
+  # the canonical "interfaces" list format, so we cannot do it any sooner than this point.
+  #
+  def link_pre_link_transform(self, link: Box, topology: Box) -> None:
+    if 'vlan' not in link:
+      return
+
+    for intf in link.interfaces:
+      vname = intf.get('vlan.access',None)
+      if not vname:
+        continue
+      if vname not in topology.nodes[intf.node].get('vlans',{}):
+        continue
+      vdata = topology.nodes[intf.node].vlans[vname]
+      for attr in topology.defaults.vlan.attributes.copy_vlan_to_intf:
+        if attr not in vdata:
```

Comment:
if attr in vdata:
   ... copy it ...

Reply:
Yeah, I was in an auto-pattern I guess ;) Thank you